src/GraphNodeManagerViewExtension/ViewModels/NodeViewModel.cs

```diff
@@ -366,8 +366,19 @@ private bool IsNodeEmptyList(MirrorData mirrorData)
         {
             if (mirrorData == null || !mirrorData.IsCollection) return false;
 
-            var list = mirrorData.GetElements();
-            return !list.Any();
+            try
+            {
+                var list = mirrorData.GetElements();
+                return !list.Any();
+            }
+            catch(NullReferenceException)
```

Comment:
The above null check for `mirrorData ` did not resolve this? I believe if `mirrorData ` was null it would have returned false from line 367.

Reply:
Actually, no. It turns out that ``mirrorData`` still throws a NullReferenceException even when it's not **null**.  It happens here 
![array_null](https://user-images.githubusercontent.com/5354594/173633004-77c981e1-f07b-4e7a-8068-4f62c5d6b143.png)
-> the ``array`` can still be null.

Reply:
Do you think I should, as part of this PR, make changes to the ``MirrorData`` class inside the ``GetElements()`` method to make sure this case is covered?

Reply:
I think you can just also add a check for MirrorData.IsNull() this is different than checking if the mirrorData wrapper is null, it checks the stack value.

Reply:
I see that we already check for `IsCollection` inside the `GetElements` method, we should also include the appropriate null check inside it as well, and then just use the output from it. This will also cover any future issues we may get into while using this class.

Reply:
@mjkkirschner @zeusongit I just pushed the commit with the alternative solution we have discussed for your review. Let me know if you think that's not a good way of handling things and I will amend accordingly.